lib/dao.js

```diff
@@ -2407,6 +2577,190 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+DataAccessObject.prototype.replaceAttributes = function(data, options, cb) {
+  var Model = this.constructor;
+  var id = getIdValue(this.constructor, this);
+  var inst = this;
+  var strict = this.__strict;
+  return Model.replaceById(id, inst, Model, data, options, strict, cb);
+};
+
+DataAccessObject.replaceById = function(id, inst, Model, data, options, strict, cb) {
```

Comment:
@bajtos: I guess I'm not following what you have in your mind based on [this task](https://github.com/strongloop/loopback-datasource-juggler/issues/825); in other words, I guess I should not include `inst`, `Model` and even `strict` in `replaceById` signature; but if I don't pass them as the method arguments, how would I access them (especially `inst`) in `replaceById` which is not a prototype method? For instance `instance` is required in `before save` in `replaceById`

If that means I need to change the implementations in connectors as well and do it like a query with `id` to replace tuple(s)... that seems a big change... 

Could you please clarify this more?

Thanks :-)